### ISSUE TITLE
object-curly-spacing warns in export statement without any braces

### ISSUE THREAD

**Issue by reintroducing** [eslint/eslint#11953]
**Tell us about your environment**

* **ESLint Version:** 6.0.1
* **Node Version:** 10.3.0
* **npm Version:** 6.1.0

**What parser (default, Babel-ESLint, etc.) are you using?**
babel-eslint

**Please show your full configuration:**
The config is comprised of multiple files, but I'll paste the offending rule below. This was working correctly before the upgrade to 6.0.1 (went from 5.16.0).

<details>
<summary>Configuration</summary>

```js
'object-curly-spacing': 2,
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
export AnimationUtils from './animation/animation';
export APIUtils from './api/api';
export BrandUtils from './brand/brand';
export DateUtils from './date/date';
export DOMUtils from './dom/dom';
export EnvironmentUtils from './environment/environment';
export MixpanelUtils from './mixpanel/mixpanel';
export NumberUtils from './number/number';
export ObjectUtils from './object/object';
export OptimizelyUtils from './optimizely/optimizely';
export RouterUtils from './router/router';
export SegmentUtils from './segment/segment';
export SentryUtils from './sentry/sentry';
export StorageUtils from './storage/storage';
export StringUtils from './string/string';
export TimeUtils from './time/time';
export ValidationUtils from './validation/validation';
```

```bash
./node_modules/.bin/eslint src/index.js
```

**What did you expect to happen?**
No errors.

**What actually happened? Please include the actual, raw output from ESLint.**
```bash
/Users/mprzybylski/Work/SpotHero/fe-utils/src/index.js
   1:1   error  There should be no space after 'export'  object-curly-spacing
   1:23  error  There should be no space before 'from'   object-curly-spacing
   2:1   error  There should be no space after 'export'  object-curly-spacing
   2:17  error  There should be no space before 'from'   object-curly-spacing
   3:1   error  There should be no space after 'export'  object-curly-spacing
   3:19  error  There should be no space before 'from'   object-curly-spacing
   4:1   error  There should be no space after 'export'  object-curly-spacing
   4:18  error  There should be no space before 'from'   object-curly-spacing
   5:1   error  There should be no space after 'export'  object-curly-spacing
   5:17  error  There should be no space before 'from'   object-curly-spacing
   6:1   error  There should be no space after 'export'  object-curly-spacing
   6:25  error  There should be no space before 'from'   object-curly-spacing
   7:1   error  There should be no space after 'export'  object-curly-spacing
   7:22  error  There should be no space before 'from'   object-curly-spacing
   8:1   error  There should be no space after 'export'  object-curly-spacing
   8:20  error  There should be no space before 'from'   object-curly-spacing
   9:1   error  There should be no space after 'export'  object-curly-spacing
   9:20  error  There should be no space before 'from'   object-curly-spacing
  10:1   error  There should be no space after 'export'  object-curly-spacing
  10:24  error  There should be no space before 'from'   object-curly-spacing
  11:1   error  There should be no space after 'export'  object-curly-spacing
  11:20  error  There should be no space before 'from'   object-curly-spacing
  12:1   error  There should be no space after 'export'  object-curly-spacing
  12:21  error  There should be no space before 'from'   object-curly-spacing
  13:1   error  There should be no space after 'export'  object-curly-spacing
  13:20  error  There should be no space before 'from'   object-curly-spacing
  14:1   error  There should be no space after 'export'  object-curly-spacing
  14:21  error  There should be no space before 'from'   object-curly-spacing
  15:1   error  There should be no space after 'export'  object-curly-spacing
  15:20  error  There should be no space before 'from'   object-curly-spacing
  16:1   error  There should be no space after 'export'  object-curly-spacing
  16:18  error  There should be no space before 'from'   object-curly-spacing
  17:1   error  There should be no space after 'export'  object-curly-spacing
  17:24  error  There should be no space before 'from'   object-curly-spacing

✖ 34 problems (34 errors, 0 warnings)
  34 errors and 0 warnings potentially fixable with the `--fix` option.
```

**Are you willing to submit a pull request to fix this bug?**
I'm not familiar enough with the project to comfortably do so.

**Comment by platinumazure** [comment]
Hi @reintroducing, thanks for the issue.

Can you recreate this issue with the default parser? This seems like it could be an issue with babel-eslint.

**Comment by reintroducing** [comment]
@platinumazure I tried doing so but then I get the following error:
```
/Users/mprzybylski/Desktop/eslint-test/index.js
  1:1  error  Parsing error: The keyword 'export' is reserved

✖ 1 problem (1 error, 0 warnings)
```

If I alter that, then I won't have the possibility of writing the statement that actually causes the issue.

I've gone ahead and set up a small test project here that demonstrates the issue for easy reproduction and simplified my config to just whats necessary.

[eslint-test.zip](https://github.com/eslint/eslint/files/3363816/eslint-test.zip)

**Comment by reintroducing** [comment]
Of course I uploaded the already edited one, my apologies. Use this zip instead.

[eslint-test 2.zip](https://github.com/eslint/eslint/files/3363827/eslint-test.2.zip)

**Comment by reintroducing** [comment]
@platinumazure Thank you for pointing me towards babel-eslint. I did some further research and found https://github.com/eslint/eslint/issues/5887 which solved this issue. I'm not sure why pre-update to 6.x worked, I'd have expected to see this issue there as well, but this makes sense and I'm going to call this one done. Thanks again and apologies for the noise.

**Comment by platinumazure** [comment]
@reintroducing If you use the default parser, you would need to provide parserOptions to get support for module parsing:

```jsonc
{
    "parserOptions": {
        "ecmaVersion": 2015,   // or higher
        "sourceType": "module"
    }
}
```

That said, you may need babel-eslint for other reasons.
